spec/lib/rake/compiler_config_spec.rb

```diff
@@ -0,0 +1,54 @@
+require File.expand_path(File.dirname(__FILE__) + '/../../spec_helper')
+
+require 'rake/extensiontask'
+require 'rbconfig'
+require 'tempfile'
+
+describe Rake::CompilerConfig do
+  def config_file(contents)
+    Tempfile.new.tap do |tf|
+      tf.write(contents)
+      tf.close
+    end
+  end
+
+  it "returns the matching config for exact platform match" do
+    cc = Rake::CompilerConfig.new(config_file(<<~EOY))
```

Comment:
Could you use `CONFIG` or something that explains content instead of `EOX` style?

```suggestion
    cc = Rake::CompilerConfig.new(config_file(<<~CONFIG))
```